compiler/src/dotty/tools/dotc/core/tasty/TreePickler.scala

```diff
@@ -695,7 +695,9 @@ class TreePickler(pickler: TastyPickler) {
             writeNat(idx)
             pickleType(tree.tpe, richTypes = true)
             args.foreach { arg =>
-              if arg.isType then writeByte(EXPLICITtpt)
+              arg.tpe match
+                case _: TermRef if arg.isType => writeByte(EXPLICITtpt)
```

Comment:
For 3.3 we can "backport" this change as

```scala
arg.tpe match
  case _: TermRef if arg.isType => report.warning(em"Term reference used in type of nested quote. This quote may fail to unpickle due to a bug that cannot be fixed in 3.3.x. This issue is fixed in 3.4.0.", arg)
  case _ =>

```

Reply:
I also believe this should be an error not a warning, and if possible give a workaround that can be used in 3.3?